reactor-core/src/main/java/reactor/core/publisher/Flux.java

```diff
@@ -3333,7 +3374,7 @@ public final T blockLast(Duration timeout) {
 	 * the same key (yet element keys can repeat in the overall sequence)
 	 */
 	public final <V> Flux<T> distinctUntilChanged(Function<? super T, ? extends V> keySelector) {
-		return distinctUntilChanged(keySelector, Objects::equals);
+		return distinctUntilChanged(keySelector, equalPredicate());
```

Comment:
this equals function change is orthogonal... shelve it and isolate it in another PR / commit on master